src/accessibility/outputs.js

```diff
@@ -0,0 +1,418 @@
+ * (example: "orange ellipse location=top left area=1%") is also available.
+ *
+ * @method gridOutput
+ * @param  {Constant} [display] either LABEL or FALLBACK (Optional)
+ *
+ * @example
+ * //<div>
+ * //<code>
+ * //textOutput();
+ * //background(148, 196, 0);
+ * //fill(255, 0, 0);
+ * //ellipse(20, 20, 20, 20);
+ * //fill(0, 0, 255);
+ * //rect(50, 50, 50, 50);
+ * //</code>
+ * //</div>
+ *
+ *
+ * //<div>
+ * //<code>
+ * //let x = 0;
+ * //function draw() {
+ *   //textOutput();
+ *   //background(148, 196, 0);
+ *   //fill(255, 0, 0);
+ *   //ellipse(x, 20, 20, 20);
+ *   //fill(0, 0, 255);
+ *   //rect(50, 50, 50, 50);
+ *   //ellipse(20, 20, 20, 20);
+ *   //x += 0.1;
+ * //}
+ *
+ */
+
+p5.prototype.gridOutput = function(display) {
+  p5._validateParameters('gridOutput', arguments);
+  if (txtOut === false) {
+    grOut = true;
+    if (this.canvas !== undefined) {
+      this._setDefaults();
+      this._createOutput('grOut');
+    } else {
+      throw new Error('gridOutput() should be called after canvas is created');
+    }
+  } else {
+    return;
+  }
+};
+
+//helper function returns true when accessible outputs are true
+p5.prototype._addAccsOutput = function() {
+  if (txtOut === true || grOut === true) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+//helper function that creates html structure for accessible outputs
+p5.prototype._createOutput = function(type) {
+  let cnvId = this.canvas.id;
+  let cIdT = cnvId + type;
+  if (document.getElementById(cIdT) === null) {
+    document
+      .getElementById(cnvId)
+      .insertAdjacentHTML(
+        'afterend',
+        '<section id="' + cIdT + '" class="accessibleOutput"></section>'
+      );
+    if (type === 'txtOut') {
+      let inner = this._createTextOutput(cIdT);
+      document.getElementById(cIdT).innerHTML = inner;
+    }
+    if (type === 'grOut') {
+      let inner = this._createGridOutput(cIdT);
+      document.getElementById(cIdT).innerHTML = inner;
+    }
+    //this._updateOutput();
+  }
+};
+
+//helper function that updates accessible outputs
+p5.prototype._updateOutput = function() {
+  if (JSON.stringify(ingredients) !== JSON.stringify(preIngredients)) {
+    this._updateOutput();
+  }
+  let cnvId = this.canvas.id;
+  if (txtOut === true) {
+    this._updateTextOutput(cnvId, ingredients, cnvConfig.background);
+  }
+  if (grOut === true) {
+    this._updateGridOutput(cnvId, ingredients, cnvConfig.background);
+  }
+};
+
+//helper function that resets all ingredients when background is called
+//and saves background color name
+p5.prototype._accsBackground = function(args) {
+  preIngredients = ingredients;
+  ingredients = {};
+  if (cnvConfig.backgroundRGBA !== args) {
+    //cnvConfig.backgroundRGBA = args;
+    cnvConfig.background = this._rgbColorName(args);
+  }
+};
+
+//helper function that gets fill and stroke of shapes
+p5.prototype._accscnvConfig = function(f, args) {
```

Comment:
maybe this should be `accsCnvCongig`?
also might make the code more readable to spell out canvas (ie `accsCanvasConfig`, `canvasConfig`)